nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -274,26 +274,32 @@ in
     consoleMode = mkOption {
       default = "keep";
 
-      type = types.enum [
-        "0"
-        "1"
-        "2"
-        "5"
-        "auto"
-        "max"
-        "keep"
-      ];
+      type =
+        with types;
+        either (enum [
+          "0"
+          "1"
+          "2"
+          "auto"
+          "max"
+          "keep"
+        ]) ints.unsigned;
```

Comment:
This will break configurations that set this to `"5"`.

Reply:
IMO "5" makes no sense as an explicitly allowed option, it's completely arbitrary. The ability to use it was only added recently and never made it into a stable release, so I don't think it will be much of a problem to remove it.

0 through 2 actually have [special meaning according to the systemd-boot documentation](https://www.freedesktop.org/software/systemd/man/latest/loader.conf.html#console-mode), so I don't mind that they're allowed as strings alongside the other special values.

Reply:
Sorry to jump in late on this PR but there are other devices which suffer from non native screen orientation and having the option to rotate the screen in all orientations would be beneficial.